File: aricode/sparring/challenges/02_fib/fib.py
#!/usr/bin/env python3
# Python - Challenge 02: Fibonacci
# Compute fib(10) = 55, return as exit code

import sys

def fib(n):
    if n < 2:
        return n
    return fib(n - 1) + fib(n - 2)

sys.exit(fib(10))

File: aricode/sparring/challenges/02_fib/fib.rs
// Rust - Challenge 02: Fibonacci
// Compute fib(10) = 55, return as exit code

use std::process::ExitCode;

fn fib(n: i32) -> i32 {
    if n < 2 { return n; }
    fib(n - 1) + fib(n - 2)
}

fn main() -> ExitCode {
    ExitCode::from(fib(10) as u8)
}
